src/mlpack/core/data/load_impl.hpp

```diff
@@ -410,6 +410,77 @@ bool LoadCategorical(const std::string& filename,
   return true;
 }
 
+template<typename MatType>
+bool Load(const std::vector<std::string>& filesname,
+          MatType& matrix,
+          const TextOptions& opts)
+{
+  TextOptions copyOpts(opts);
+  return Load(filesname, matrix, copyOpts);
+}
+
+template<typename MatType>
+bool Load(const std::vector<std::string>& filesname,
+          MatType& matrix,
+          TextOptions& opts)
+{
+  bool success;
+  MatType tmp;
+  arma::field<std::string> firstHeaders;
+  for (size_t i = 0; i < filesname.size(); ++i)
+  {
+    success = Load(filesname.at(i), matrix, opts); 
+    if (opts.HasHeaders())
+    {
+      if (i == 0)
+        firstHeaders = opts.Headers();
+      else
+      {
+        arma::field<std::string> headers = opts.Headers();
+        if (firstHeaders.at(0) != headers.at(0))
+          Log::Warn << "Load(): Headers from different CSV does not match!"
+              << std::endl;
+      }
+    }
+
+    if (success)
+    {
+      if (i == 0)
+      {
+        tmp = std::move(matrix);
+      }
+      else
+      {
+        if (!opts.NoTranspose()) // if transpose
+        {
+          if (tmp.n_rows != matrix.n_rows)
+          {
+            Log::Fatal << "Load(): dimension mismatch when loading vectors" << std::endl;
```

Comment:
We should check `opts.Fatal()` here to decide whether to use `Warn` or `Fatal`.  Also we should probably output the name of the failed file, to make the error message more actionable for users.

Reply:
I thought about it but it does not make sense really because you will end up with a half of the files loaded and the others not.

If there are some cases where you think that some files should be loaded and not others, then I am happy to throw a warning. For me, this looks similar to when you are loading a set of images.

In the future, this function will be the unified interface for text and images, with the only difference being the Options.